shared/wallets/transaction-details/index.tsx

```diff
@@ -428,10 +428,15 @@ class LoadTransactionDetails extends React.Component<Props> {
   componentDidUpdate(prevProps: Props) {
     // An erased transaction ID likely means the payment was updated,
     // which means details need to be retrieved again
+    if (this.props.loading || prevProps.loading) {
+      return
+    }
+    const props = this.props as NotLoadingProps
+    const prev = prevProps as NotLoadingProps
     if (
-      (!this.props.transactionID || !this.props.senderAccountID) &&
-      prevProps.transactionID &&
-      prevProps.senderAccountID &&
+      (!props.transactionID || !props.senderAccountID) &&
+      prev.transactionID &&
+      prev.senderAccountID &&
```

Comment:
Is this the best way to do this?

Reply:
You can use [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html) to refine the types. For example you can do this:
```
diff --git a/shared/wallets/transaction-details/index.tsx b/shared/wallets/transaction-details/index.tsx
index 07dd203790..1392eabcd8 100644
--- a/shared/wallets/transaction-details/index.tsx
+++ b/shared/wallets/transaction-details/index.tsx
@@ -421,6 +421,10 @@ const TransactionDetails = (props: NotLoadingProps) => {
   )
 }
 
+function isNotLoadingProps(props: Props): props is NotLoadingProps {
+  return !props.loading
+}
+
 class LoadTransactionDetails extends React.Component<Props> {
   componentDidMount() {
     this.props.onLoadPaymentDetail()
@@ -428,11 +432,11 @@ class LoadTransactionDetails extends React.Component<Props> {
   componentDidUpdate(prevProps: Props) {
     // An erased transaction ID likely means the payment was updated,
     // which means details need to be retrieved again
-    if (this.props.loading || prevProps.loading) {
+    if (!isNotLoadingProps(this.props) || !isNotLoadingProps(prevProps)) {
       return
     }
-    const props = this.props as NotLoadingProps
-    const prev = prevProps as NotLoadingProps
+    const props = this.props
+    const prev = prevProps
     if (
       (!props.transactionID || !props.senderAccountID) &&
       prev.transactionID &&
```

`as` should generally be avoided since it is telling the compiler "I know better than you, trust me on this."